cli/src/index.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import * as process from 'process'
+
+import { render } from 'ink'
+import yargs from 'yargs/yargs'
+import type { Argv, BuilderCallback } from 'yargs'
+import { hideBin } from 'yargs/helpers'
+import options, { commandOptions } from './options.ts'
+import { CommandKeys, Commands } from './commands.tsx'
+import Mainframe from './mainframe.tsx'
+import { buildPublicClient, parseInitialNetwork } from './client.tsx'
+
+const main = (): Argv => {
+  let cmd = yargs(hideBin(process.argv))
+    .middleware(async (arg) => {
+      try {
+        const network = parseInitialNetwork(arg)
+        const client = buildPublicClient(network)
+
+        console.log('Testing RPC connection...')
+
+        if (!client) {
+          console.error('Failed to create RPC client!')
+          process.exit(1)
+        }
+
+        const resChainId = await client.getChainId()
+
+        if (network.chainId !== resChainId) {
+          console.error(
+            `Chain ID mismatch! RPC Response: ${resChainId}, Expected: ${network.chainId}`
+          )
+          process.exit(1)
+        }
+        console.log(
+          'RPC Connection test completed. Retrieving option pool stats...'
+        )
+      } catch (e) {
+        console.error(e.message)
+        process.exit(1)
+      }
+    })
+    .command(
+      '$0',
+      'Interactively communicate with a Panoptic option pool deployment. Configuration parameters (network, contract address, wallet...) can be set using command line arguments, or inside the interactive interface\n'
+    )
+    .command(
+      'start',
+      'Enter interactive mode\n',
+      () => {},
+      async (arg) => {
+        const { waitUntilExit } = render(<Mainframe options={arg} />)
```

Comment:
Instead of rendering mainframe in this file (which is originally a ts, outside the context of react), it can be rendered in separate tsx files within React context, and moreover rendered with more flexibility (e.g. just render once and exit, or render subcomponents instead of the whole thing, or with different configs) instead of always "waitUntilExit" and only the whole MainFrame

Reply:
- Moved mainframe rendering from `index` to `mainframe.tsx`
- I don't think rendering only subcomponent is possible, because it depends on wrapping providers